packages/react/src/app-layout/AppLayoutSidebarNav.test.tsx

```diff
@@ -0,0 +1,145 @@
+import { ReactNode } from 'react';
+import { render, screen } from '../../../../test-utils';
+import { AppLayoutContext } from './AppLayoutContext';
+import { AppLayoutSidebarNav } from './AppLayoutSidebarNav';
+import { navigationItems } from './test-utils';
+
+const NAV_ITEMS = navigationItems('Antfix');
+const ALL_LINKS = [
+	'/account',
+	'/',
+	'/establishments',
+	'/establishments/canberra',
+	'/establishments/sydney',
+	'/intelligence',
+	'/compliance',
+	'/compliance/audit',
+	'/compliance/certificates',
+	'/help',
+];
+const LEVEL_1_LINKS = ALL_LINKS.filter(
+	(link) => link.match(/\//g)?.length === 1
+);
+const LEVEL_2_LINKS = ALL_LINKS.filter(
+	(link) => link.match(/\//g)?.length === 2
+);
+
+const ContextWrapper = ({ children }: { children: ReactNode }) => (
+	<AppLayoutContext.Provider
+		value={{
+			focusMode: false,
+			isMobileMenuOpen: true,
+			openMobileMenu: console.log,
+			closeMobileMenu: console.log,
+		}}
+	>
+		{children}
+	</AppLayoutContext.Provider>
+);
+
+describe('Given AppLayoutSidebarNav is rendered', () => {
+	describe('when subLevelVisible is "whenActive"', () => {
+		describe('when the active item has no sub-level items', () => {
+			test('then no sub-level items should be visible', async () => {
+				render(
+					<ContextWrapper>
+						<AppLayoutSidebarNav
+							activePath="/"
+							items={NAV_ITEMS}
+							subLevelVisible="whenActive"
+						/>
+					</ContextWrapper>
+				);
+
+				const navItems: HTMLAnchorElement[] = await screen.findAllByRole(
+					'link'
+				);
+				const navItemPathnames = navItems.map(
+					(item) => new URL(item.href).pathname
+				);
+
+				expect(navItemPathnames).toEqual(LEVEL_1_LINKS);
```

Comment:
Should we also be testing which has the actiev stuff applied? Perhaps just some snapshots to ensure the markup is good?

Reply:
Ideally this is visual regression testing as a snapshot will only show the classname change, I believe. I can add one, though.

- [ ] snapshot for active styles